docs/reference/api/connectivity/networksocket/DNS.md

```diff
@@ -0,0 +1,90 @@
+## DNS Resolver
+
+The DNS resolver provides an interface to do DNS host name resolutions.
```

Comment:
Please expand on when and why you'd want to use this API.

Reply:
Added more description about dsn.